qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/QosRateLimiter.java

```diff
@@ -123,9 +123,14 @@ private synchronized void createRateLimiterAtomically() {
      */
     public void recordAdjustment(long adjustmentUnits) {
         if (adjustmentUnits > 0) {
-            rateLimiter.steal(Ints.saturatedCast(adjustmentUnits)); // TODO(nziebart): deal with longs
+            rateLimiter.steal(adjustmentUnits);
+        } else {
+            try {
+                rateLimiter.returnPermits(LongMath.checkedMultiply(adjustmentUnits, -1), stopwatch.readNanos());
+            } catch (ArithmeticException e) {
```

Comment:
can multiplying by -1 ever cause an issue?

Reply:
Yup, if its Long.MIN_VALUE, caught by a test. :D

Reply:
nit: move the safe multiply operation to its own method (e.g.`safeAbs`) so the `try/catch` isn't around the whole method call